data/core/doc/init.lua

```diff
@@ -325,6 +339,20 @@ function Doc:raw_insert(line, col, text, undo_stack, time)
   push_undo(undo_stack, time, "selection", unpack(self.selections))
   push_undo(undo_stack, time, "remove", line, col, line2, col2)
 
+  if #lines > 1 then
+    -- Need to shift all the subsequent long lines
+    local line_numbers = {}
```

Comment:
I think we may avoid recreating a new table here for line_numbers.

Reply:
How should we do that? We need to modify entries in the table that follow the current one.